proto/kuksa/val/v1/val.proto

```diff
@@ -98,6 +98,7 @@ message SubscribeEntry {
 // Subscribe to changes in datapoints.
 message SubscribeRequest {
   repeated SubscribeEntry entries = 1;
+  optional uint64 frequency_hertz = 2;
```

Comment:
Some thoughts

- Do we see Hertz and uint64 as a reasonable representation. That makes it impossible to support intervals longer than 1 second. An alternative solution could be to for example use `interval_ms`, i.e. wanted interval in milliseconds. That would allow us to represent both 100 Hz as well as 0.1 Hz
- Should we possible already in this file describe semantics? That would be good if we later intend to promote this API to a "standardized" API. I.e. document topics like
  -  What is default if this field/member is not given? Max frequency? On change? Other?
  - What shall happen if 0 as given as frequency? Does that imply "on change" or does it just mean "as fast as possible, changed or not"
  - What is the semantic meaning of specifying frequency - is it getting latest value at that frequency regardless of whether the value has been updated (new timestamp) or changed (changed value)
  - What shall happen if frequency/interval cannot be supported. Like if you request 2000 Hz - should you get an error?

Reply:
I have completely missed that frequency could range from 0 to 1 as well, so uint64 is totally wrong, better to use float type (will push the fix). I would still consider to use frequency hertz representation better than interval_ms since hertz is also used by VHAL interface for Continuous properties -> https://source.android.com/docs/automotive/vhal/vhal-interface

Regarding your questions:

> 1. What is default if this field/member is not given? Max frequency? On change? Other?

- The initial default behaviour implemented works as follow: 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;1. if the Subscribe request contains **any** continuous signal without a frequency parameter it returns an _InvalidArgument_ error.
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;2. Same in case the request doesn't contain **any** continuous signal and a frequency parameter is given it must return an _InvalidArgument_ error.

> 2. What shall happen if 0 as given as frequency? Does that imply "on change" or does it just mean "as fast as possible, changed or not"
- No behaviour specified, I would suggest to return an _InvalidArgument_ error as well since it would mean to divide some number by 0. What do you think?

> 3. What is the semantic meaning of specifying frequency - is it getting latest value at that frequency regardless of whether the value has been updated (new timestamp) or changed (changed value)

- The semantic meaning is to get latest value at that frequency regardless of whether the value has been updated or changed with its latest update or change timestamp.

> 4. What shall happen if frequency/interval cannot be supported. Like if you request 2000 Hz - should you get an error?

- We should discuss this again, we can suggest the customer to specific its own frequency limit rate at broker initialization (should to be implemented) depending on its hardware capabilities.

Reply:
I think it could be good to have a configurable "max frequency", similar to the "max number of signals" topic we discussed as part of the Audit findings. Then one could possible let the frequency 0 mean "as fast as possible" rather than considering it as an illegal value.

Anyway, maybe we should document this with comments so we could generate documentation with frameworks like https://github.com/pseudomuto/protoc-gen-doc so that expected behavior is clearly defined

Reply:
Findings:
* Regardless, if we use "frequency" or "interval": It should be already clear from the parameter name, what it means. Currently, I cannot see (without reading the documentation) if it is minimum, typical, or maximum frequency/interval.
* Don't we add detailed documentation about the behaviour in the proto files? I agree with Erik: We should also describe the semantics in the interface definition. Imho, the "outside visible behaviour" always belongs to the contract description of an interface definition.
* In the current definition this parameter is related to the overall subscription, but not to certain signals - which is fine for me if the parameter defines the max update frequency. But the behaviour should be clearly defined if the the subscription comprises signals with different update frequencies.
* What is the behaviour, if signals are updated faster than the max frequency requested by the client: 
  * Does it get "just" some sample value at the moment the update interval is reached,
  * do we send some mean value of the last samples (how would that be calculated?),
  * do we accumulated values and send them as a batch,
  * ...?
If we are unsure about the behaviour to be achieved by this PR, we should rethink merging it ...